web/src/test/java/org/cbioportal/web/PatientControllerTest.java

```diff
@@ -0,0 +1,243 @@
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudyId").value(TEST_CANCER_STUDY_IDENTIFIER_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudyIdentifier").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudy").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].internalId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].stableId").value(TEST_STABLE_ID_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudyId").value(TEST_CANCER_STUDY_IDENTIFIER_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudyIdentifier").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudy").doesNotExist());
+    }
+
+    @Test
+    public void getAllPatientsInStudyMetaProjection() throws Exception {
+
+        BaseMeta baseMeta = new BaseMeta();
+        baseMeta.setTotalCount(2);
+
+        Mockito.when(patientService.getMetaPatientsInStudy(Mockito.anyString())).thenReturn(baseMeta);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/studies/test_study_id/patients")
+            .param("projection", "META"))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.header().string(HeaderKeyConstants.TOTAL_COUNT, "2"));
+    }
+
+    @Test
+    public void getPatientInStudyNotFound() throws Exception {
+
+        Mockito.when(patientService.getPatientInStudy(Mockito.anyString(), Mockito.anyString())).thenThrow(
+            new PatientNotFoundException("test_study_id", "test_patient_id"));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/studies/test_study_id/patients/test_patient_id")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isNotFound())
+            .andExpect(MockMvcResultMatchers.jsonPath("$.message")
+                .value("Patient not found in study test_study_id: test_patient_id"));
+    }
+
+    @Test
+    public void getPatientInStudy() throws Exception {
+
+        Patient patient = new Patient();
+        patient.setInternalId(TEST_INTERNAL_ID_1);
+        patient.setStableId(TEST_STABLE_ID_1);
+        patient.setCancerStudyId(TEST_CANCER_STUDY_ID_1);
+        patient.setCancerStudyIdentifier(TEST_CANCER_STUDY_IDENTIFIER_1);
+        CancerStudy cancerStudy = new CancerStudy();
+        cancerStudy.setCancerStudyId(TEST_CANCER_STUDY_ID_1);
+        cancerStudy.setCancerStudyIdentifier(TEST_CANCER_STUDY_IDENTIFIER_1);
+        cancerStudy.setTypeOfCancerId(TEST_TYPE_OF_CANCER_ID_1);
+        cancerStudy.setName(TEST_NAME_1);
+        cancerStudy.setShortName(TEST_SHORT_NAME_1);
+        cancerStudy.setDescription(TEST_DESCRIPTION_1);
+        patient.setCancerStudy(cancerStudy);
+
+        Mockito.when(patientService.getPatientInStudy(Mockito.anyString(), Mockito.anyString())).thenReturn(patient);
```

Comment:
one patient return value .. construction could also be externalized to be side-by-side with the two patient constructor (encourage reuse) but it is fine here too